### ISSUE TITLE
Replace ginkgo runner with testify assertions

### ISSUE THREAD

**Issue by chayim** [redis/go-redis#2896]

**Comment by git-hulk** [comment]
@chayim Do you mean to remove ginkgo and replace it with testify assertions? If yes, it needs massive work to refactor old test cases, but we can split it into sub-tasks and welcome other contributors to help.

I will be happy to take this issue if no one is working on it now.

**Comment by git-hulk** [comment]
@ndyakov Yes, I prefer using testify over ginkgo. The main benefit is that we could easily run a single directly from the IDE when debugging.

**Comment by ndyakov** [comment]
I agree that ginkgo is not great, but the effort to change it is significant. If you are up to the task, go at it.

**Comment by ndyakov** [comment]
we are planning to work on this in the following months, feel free to add suggestions and ideas here. @ofekshenawa will own this initiative.
